sycl/test/extensions/properties/properties_reduction.cpp

```diff
@@ -0,0 +1,28 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+
+#include <sycl/sycl.hpp>
+
+int main() {
+  int *r = nullptr;
+  // Must not use `sycl_ext_oneapi_reduction_properties`'s overloads:
+  std::ignore =
+      sycl::reduction(r, sycl::plus<int>{},
+                      sycl::property::reduction::initialize_to_identity{});
+
+  namespace sycl_exp = sycl::ext::oneapi::experimental;
+  std::ignore =
+      sycl::reduction(r, sycl::plus<int>{},
+                      sycl_exp::properties(sycl_exp::initialize_to_identity));
```

Comment:
Should we augment the test for `deterministic` property as well?

Reply:
Why? The change here is about "negative" behavior, this particular line acts less like a test/check and more like "context"/"back-reference". Positive behavior tests are expected to have been added in the original PR introducing these properties.

Reply:
I see. Looks good then. Also, please update the PR description to summarize the changes in this PR.

Reply:
I think title already says it all. What else do you want to see there?

Reply:
It is not clear from the title what "Constraints"/"Negative behavior" you are referring to.

Reply:
Can you provide the text for me to copy-paste? I think "Constraints" is unambiguous when applied to the SYCL interfaces/extensions.

Reply:
How about something like "This PR ensures that `sycl::reduction` is visible only when we use a property list, which is an instance of `sycl::ext::oneapi::experimental::properties`, containing only `deterministic` and/or `initialize_to_identity`."

>  I think "Constraints" is unambiguous when applied to the SYCL interfaces/extensions.

A SYCL extension can have many constraints over different functions. Which constraints did you enforce in this PR?

Reply:
> A SYCL extension can have many constraints over different functions. Which constraints did you enforce in this PR?

"constraints from sycl_ext_oneapi_reduction_properties"; all of them AFAIK.

> "This PR ensures that sycl::reduction is visible only when we use a property list, which is an instance of sycl::ext::oneapi::experimental::properties, containing only deterministic and/or initialize_to_identity."

Why do you want to quote the entire constraint from the extension in the PR message? What would you do if different methods had slightly different constraints? Would you list all of them separately? What happened to Don't Repeat Yourself principle? IMO, that's just marginally more useful than comments of this nature:

```
// Provide custom `operator==` overload
bool operator==(const S &other) const {}
```

Reply:
> all of them AFAIK

There we go. *All constraints* is the keyword I'm looking for.

> What happened to Don't Repeat Yourself principle? IMO, that's just marginally more useful than comments of this nature

I think this analogy is incorrect: I'm not asking for redundant comments in code but rather a descriptive commit message. What's the downside of quoting the constraints that you implemented from extension in the PR description as well? Imagine someone trying to compiler release notes and coming across this PR, a descriptive commit message would save them the effort of reading the extension and going over code changes in this PR to figure out which constraints you implemented.

Reply:
Anyway, that's just my opinion. If folks from @intel/llvm-gatekeepers feel otherwise, please feel free to merge the PR.

Reply:
> There we go. _All constraints_ is the keyword I'm looking for.

That's probably language barrier :) In the absence of "some" I'd just assume "all" by default :)

> What's the downside of quoting the constraints...

You haven't answered my question:

> What would you do if different methods had slightly different constraints? Would you list all of them separately?

I can't imagine the answer for that would be copy-pasting them all into this PR. And if we agree that we wouldn't do that for such more complex case, I can't see how a simpler one deserves a longer PR description.

Reply:
> What would you do if different methods had slightly different constraints? Would you list all of them separately?

Well, if the constraints were to be slightly different, we could still try to summarize them, without being overly specific.
To quote LLVM's developer policy on commit messages (https://llvm.org/docs/DeveloperPolicy.html#commit-messages):

> Most importantly, the contents of the message should be carefully written to convey the rationale of the change (without delving too much in detail). It also should avoid being vague or overly specific.

Reply:
> we could still try to summarize them

That would be "implement constraints from <extension>", wouldn't it?

Reply:
> That would be "implement constraints from ", wouldn't it?

Depends on how different the constraints are. If they are just "slightly" different, I would have come up with a better commit message than "implement constraints". If they are very different, to the extent of being unrelated, I would have tried to break PRs into smaller ones.

For the record, this is not a blocker from me since changes in SYCL RT looks good. So, if you feel strongly about not adding a commit message, please feel free to proceed with the merge.

Reply:
> So, if you feel strongly about not adding a commit message, please feel free to proceed with the merge.

I feel strongly about [NOT] quoting the spec :)